src/Microsoft.Security.Utilities.Core/SecretMasker.cs

```diff
@@ -142,6 +142,8 @@ public string MaskSecrets(string input)
         Detection? currentDetection = default;
         foreach (Detection detection in detections.OrderBy(x => x.Start))
         {
+            onDetection?.Invoke(detection);
```

Comment:
>onDetection [](http://example.com/codeflow?start=12&length=11)

OK, so we don't assume a need to provide any synchronization here, that would be up to action provider to handle if necessary.

You didn't create all the debt in this file, but I wonder if we should add a doc comment for this method and maybe start addressing some clean-up as we touch various things.

This file is in an unfortunately inconsistent condition but we do have some other doc xml. Maybe just add a good summary for this changed method?

Reply:
Sure, I'll add a doc.

Reply:
On synchronization, yes, but now I'm again annoyed that Detection is mutable. Maybe it's time to bite that bullet. If detection were mutated here, it could break the algorithm.

Reply:
OK, I bit the bullet, see updated title description and code changes.